apps/src/applab/appStorage.js

```diff
@@ -15,67 +15,72 @@ AppStorage.tempEncryptedAppId =
  * @param {string} record.tableName The name of the table to read from.
  * @param {Object} record Object containing other properties to store
  *     on the record.
- * @param {Function} callback Function to call with the resulting record.
+ * @param {function(Object)} onSuccess Function to call with the new record.
+ * @param {function(string)} onError Function to call with an error message
+ *    in case of failure.
  */
-AppStorage.createSharedRecord = function(record, callback) {
+AppStorage.createSharedRecord = function(record, onSuccess, onError) {
   var tableName = record.tableName;
   if (!tableName) {
-    console.log('readRecords: missing required property "tableName"');
+    onError('error reading records: missing required property "tableName"');
```

Comment:
nevermind, I see now that you don't actually allow the student to pass in an onError callback. Which is mostly fine, but does prevent them from being able to write code that behaves differently on failure.